android/src/main/java/com/polidea/blemulator/SimulatedAdapter.java

```diff
@@ -257,6 +261,13 @@ public void onSuccess(DeviceContainer deviceContainer) {
                     deviceContainer.setConnected(oldContainer.isConnected());
                 }
                 knownPeripherals.put(deviceContainer.getIdentifier(), deviceContainer);
+
+                for (List<CharacteristicContainer> characteristicContainers : deviceContainer.getCharacteristicContainersIndexedByServiceUuids().values()) {
+                    for (CharacteristicContainer characteristicContainer : characteristicContainers) {
+                        knownCharacteristics.put(characteristicContainer.getCharacteristic().getId(), characteristicContainer);
```

Comment:
Not directly related to this PR but is it certain that `id` of each characteristic is unique in BLEmulator if characteristics use the same `UUID`?

Reply:
Yes, BLEmulator uses incrementing counter instead of hashing.